lib/srv/alpnproxy/local_proxy.go

```diff
@@ -303,3 +316,92 @@ func (l *LocalProxy) Close() error {
 	}
 	return nil
 }
+
+// StartAWSAccessProxy starts the local AWS CLI proxy.
+func (l *LocalProxy) StartAWSAccessProxy(ctx context.Context) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			NextProtos:         []string{string(l.cfg.Protocol)},
+			InsecureSkipVerify: l.cfg.InsecureSkipVerify,
+			ServerName:         l.cfg.SNI,
+			Certificates:       l.cfg.Certs,
+		},
+	}
+	proxy := &httputil.ReverseProxy{
+		Director: func(outReq *http.Request) {
+			outReq.URL.Scheme = "https"
+			outReq.URL.Host = l.cfg.RemoteProxyAddr
+		},
+		Transport: tr,
+	}
+	err := http.Serve(l.cfg.Listener, http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		if err := l.verifyAWSSignature(req); err != nil {
+			rw.WriteHeader(http.StatusForbidden)
+			return
+		}
+		proxy.ServeHTTP(rw, req)
+	}))
+	if err != nil && !utils.IsUseOfClosedNetworkError(err) {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// verifyAWSSignature verifies the request signature ensuring that the request originates from tsh aws command execution
+// AWS CLI signs the request with random generated credentials that are passed to LocalProxy by
+// the AWSCredentials LocalProxyConfig configuration.
+func (l *LocalProxy) verifyAWSSignature(req *http.Request) error {
```

Comment:
What do you think about factoring this out into a some kind of service in `lib/srv/app/aws`? Along with `filterSignedHeaders`. Just feels like local proxy has too much specifics instead of relying on services.